xGame/Java/xgame/src/test/java/com/xgame/restservice/ChessMatchService.java

```diff
@@ -16,33 +27,112 @@
+			assertEquals(newMatch.getBlackEmail(), player2.getEmail());
+			assertEquals(newMatch.getTurnCount(), 0);
+		}
+		catch(Exception e) {
+			fail();
+		}
+		finally {
+			//cleanup
+			matchRepo.deleteById(newMatch.getId());
+			userRepo.delete(player1);
+			userRepo.delete(player2);
+		}
+		
+	}
+	
+	@Test
+	void getInvites() {
+		var player1 = userRepo.save(new User("junit1", "junit1@email.com", "junit1password"));
+		var player2 = userRepo.save(new User("junit2", "junit2@email.com", "junit2password"));
+		
+		var chessMatch = new ChessMatch("This is a test board");
+		chessMatch.setWhitePlayer(player1);
+		chessMatch.setBlackPlayer(player2);
+		chessMatch.setTurnCount(0);
+		chessMatch.setMatchStatus(MatchStatus.PENDING);
+		
+		var match = matchRepo.save(chessMatch);
 		
-		assertTrue(userCount >= 2, "There are not enough Users to create a match.");
+		try {
+				
+		//test invites
+		var invites = service.getInvites(player2.getId());
+		assertEquals(invites.size(), 1);
 		
-		var users = userRepo.findAll();
-		var user1 = users.get(0);
-		var user2 = users.get(1);
+		var invite = invites.get(0);
+		assertEquals(invite.getWhitePlayerNickname(), "junit1");
+		assertEquals(invite.getMatchId(), match.getId());
+		}
+		catch(Exception e) {
+			fail();
+		}
+		finally {
+			//cleanup
+			matchRepo.delete(match);
+			userRepo.delete(player1);
+			userRepo.delete(player2);
+		}
+	}
+	
+	@Test
+	void acceptInvite() {
 		
-		var newMatch = service.createMatch(user1.getId(), user2.getId());
+		var player1 = userRepo.save(new User("junit1", "junit1@email.com", "junit1password"));
+		var player2 = userRepo.save(new User("junit2", "junit2@email.com", "junit2password"));
+		List<Message> messages = new ArrayList<Message>();
 		
-		assertNotNull(newMatch);
-		assertEquals(matchCount + 1, matchRepo.count());
-		assertEquals(newMatch.getWhiteId(), user1.getId());
-		assertEquals(newMatch.getBlackId(), user2.getId());
-		assertEquals(newMatch.getWhiteEmail(), user1.getEmail());
-		assertEquals(newMatch.getBlackEmail(), user2.getEmail());
-		assertEquals(newMatch.getTurnCount(), 0);
+		var chessMatch = new ChessMatch("This is a test board");
+		chessMatch.setWhitePlayer(player1);
+		chessMatch.setBlackPlayer(player2);
+		chessMatch.setTurnCount(0);
+		chessMatch.setMatchStatus(MatchStatus.PENDING);
 		
-		//cleanup
-		matchRepo.deleteById(newMatch.getMatchId());
+		var match = matchRepo.saveAndFlush(chessMatch);
+		try {
+			//test match acceptance
+			var success = service.acceptInvite(match.getId());
+			assertTrue(success);
+			
+			messages = messageRepo.findByUserId(player1.getId());
+			assertEquals(messages.size(), 1);
+			assertEquals(messages.get(0).getContents(), player2.getNickname() + " has accepted your invitation to a match!");
+			
+			//match = matchRepo.getOne(match.getId());
+			//assertEquals(match.getMatchStatus(), MatchStatus.INPROGRESS);
+			//assertTrue(match.getStartTimestamp() != null);
```

Comment:
Do we need these commented lines?

Reply:
I was trying to write a more robust test but I couldn't get it to work so I left it incase I wanted to come back and implement it later.